qbraid_qir/qasm3/visitor.py

```diff
@@ -91,7 +97,7 @@ def __init__(self, initialize_runtime: bool = True, record_output: bool = True):
         self._module = None
         self._builder = None
         self._entry_point = None
-        self._scope = Scope.GLOBAL
+        self._scope = [{}]
```

Comment:
Using a deque (double-ended queue) from the collections module can sometimes be more performant than lists for operations involving elements insertion and removal at both ends. This is because deque is optimized for fast fixed-time operations at both ends. For managing scope in a stack-like manner (where only the end operations matter), the performance difference might be negligible, but if we expect a high frequency of push and pop operations, then we should probably consider using deque.

```python
from collections import deque

...

self._scope = deque([{}])
```